### PR TITLE
Buckets page cypress tests

### PR DESCRIPTION
**Pull request by SanjalKatiyar**

### CODE REVIEW

File: cypress/views/common.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { DATA_FEDERATION, DATA_SERVICES } from '../constants/common';
+
+export const MCGMSCommon = {
+  visitMCGMSDashboard: () => {
```
Comment (debjyoti-pandit) [inline cypress/views/common.ts:4]: this is a standard mcg-mc-console operation we can keep in the separate mcg-ms console file. 

the remaining are related general console which is other than our operator.

what you say, should we keep a separate these?

File: cypress/tests/bucket-policy-spec.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  // right now we are only creating "nsfs" (FileSystem) type NamespaceStore which is only allowed with BucketClass of type "Single".
+  it('creates Bucket policy with single data source', () => {
+    cy.exec(
+      `echo '${JSON.stringify(
+        dataSourceNSFS(dataSourceName, pvcName, 'e2e-subPath')
+      )}' | oc create -f -`
+    ).then(() => {
+      cy.byTestID('item-create').click();
+      cy.byTestID('bucket-name-text').type(singleBucketPolicy);
+      cy.byTestID('read-write-dropdown')
```
Comment (debjyoti-pandit) [inline cypress/tests/bucket-policy-spec.ts:53]: image

failing

File: cypress/tests/bucket-policy-spec.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import { DATA_FEDERATION_NAMESPACE, SECOND, MINUTE } from '../constants/common';
```
Comment (debjyoti-pandit) [inline cypress/tests/bucket-policy-spec.ts:1]: file name - .spec.ts

File: cypress/constants/common.ts [hunk 4]
```diff
@@ -4,4 +4,7 @@ export const BRIDGE_PASSWORD = 'BRIDGE_KUBEADMIN_PASSWORD';
 
 export const DATA_SERVICES = 'Data Services'; // nav menu category option
 export const DATA_FEDERATION = 'Data Federation'; // nav menu sub category option
-export const STATUS_CARD_TITLE = 'Data Federation service';
+
+export const DATA_FEDERATION_NAMESPACE = 'redhat-data-federation';
```
Comment (debjyoti-pandit) [inline cypress/constants/common.ts:8]: let's keep the constant key as same as the value to remove confusion. REDHAT_DATA_FEDERATION seems to be good

File: cypress/tests/bucket-policy-spec.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import { dataSourceNSFS } from '../mocks/data-source';
+import { MCGMSCommon, projectNameSpace } from '../views/common';
+import { pvc } from '../views/pvc';
+
+describe('Bucket policy page', () => {
+  const singleBucketPolicy = 'e2e-bucket-single';
```
Comment (debjyoti-pandit) [inline cypress/tests/bucket-policy-spec.ts:7]: these are all constant names, should we write them in caps? what is your opinion?

Comment (debjyoti-pandit) [reply cypress/tests/bucket-policy-spec.ts:7]: or we can have a test value file form where all values will be picked up

File: cypress/tests/bucket-policy-spec.ts (hunk 2 above)
Comment (SanjalKatiyar) [reply cypress/tests/bucket-policy-spec.ts:53]: did u add the "data-id" ??

File: cypress/views/common.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    cy.byLegacyTestID('horizontal-link-Buckets').click();
+  },
+};
+
+export const app = {
+  waitForDocumentLoad: () => {
+    cy.document().its('readyState').should('eq', 'complete');
+  },
+  waitForLoad: (timeout: number = 160000) => {
+    cy.get('.co-m-loader', { timeout }).should('not.exist'); // eslint-disable-line cypress/require-data-selectors
```
Comment (debjyoti-pandit) [inline cypress/views/common.ts:23]: what are these checks for, can we have a log mentioning it?

File: cypress/views/common.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  clickProjectDropdown: () => {
+    cy.byLegacyTestID('namespace-bar-dropdown').find('button').first().click();
+  },
+
+  enterProjectName: (projectName: string) => {
+    cy.byLegacyTestID('modal-cancel-action').should('be.visible');
+    cy.get('#input-name').type(projectName); // eslint-disable-line cypress/require-data-selectors
+  },
+
+  selectOrCreateProject: (projectName: string) => {
```
Comment (debjyoti-pandit) [inline cypress/views/common.ts:42]: can we add some logs in the inter-mediatory steps

File: cypress/tests/bucket-policy-spec.ts (hunk 2 above)
Comment (debjyoti-pandit) [reply cypress/tests/bucket-policy-spec.ts:53]: where and what I need to add? I just pulled your branch and ran the tests

File: cypress/views/common.ts (hunk 1 above)
Comment (SanjalKatiyar) [reply cypress/views/common.ts]: ok

File: cypress/tests/bucket-policy-spec.ts (hunk 2 above)
Comment (SanjalKatiyar) [reply cypress/tests/bucket-policy-spec.ts]: CI and local both are passing for me...

File: cypress/tests/bucket-policy-spec.ts (hunk 3 above)
Comment (SanjalKatiyar) [reply cypress/tests/bucket-policy-spec.ts]: ack...

File: cypress/constants/common.ts (hunk 4 above)
Comment (SanjalKatiyar) [reply cypress/constants/common.ts:8]: we r using same in `package/component` as well... seems okay...

File: cypress/tests/bucket-policy-spec.ts (hunk 5 above)
Comment (SanjalKatiyar) [reply cypress/tests/bucket-policy-spec.ts]: done...

File: cypress/views/common.ts (hunk 6 above)
Comment (SanjalKatiyar) [reply cypress/views/common.ts:7]: `waitForLoad` should be clear enough... still added log anyway...

File: cypress/tests/bucket-policy-spec.ts (hunk 2 above)
Comment (SanjalKatiyar) [reply cypress/tests/bucket-policy-spec.ts]: let's wait for CI once again...

**Pull request merged**